src/plugins/_manager.ts

```diff
@@ -0,0 +1,171 @@
+
+    const pkg = this.loadPackage(options.pkg, options.version, options.installEnv ?? {});
+
+    const clazz = pkg.module[options.class];
+    if (!clazz) {
+      throw new Error(`Unable to locate class '${options.class}' in package '${options.pkg}@${options.version}'. Are you sure you exported it?`);
+    }
+
+    return { instance: new clazz(options.properties ?? {}), class: options.class, ...pkg };
+  }
+
+  private loadPackage(pkg: string, version: string, installEnv: { [key: string]: any }): Package {
+
+    if (isNaN(parseInt(version.charAt(0)))) {
+      // forbid version ranges because otherwise we need to do semver comparison and introduce a dependency.
+      // also ranges make it easier to get left behind since it might give the false impression we will be installing
+      // the latest version (which we will not because it would mean contacting NPM on every synth)
+      throw new Error(`Unsupported version spec for package ${pkg}: ${version}. Cannot be a range.`);
+    }
+
+    const proto = url.parse(pkg).protocol;
+
+    if (proto) {
+      // urls are not supported because they don't provide a name with which we can 'require' the module.
+      // if needed, we can make the loader smarter and enable this, but not for now.
+      throw new Error(`Unsupported package reference: ${pkg}. Can either be an NPM package name, or a local path to a directory`);
+    }
+
+    const local = path.isAbsolute(pkg)
+      // relative paths must start with '.' because otherise they
+      // are indistinguishable from npm package names.
+      || pkg.startsWith(`.${path.sep}`);
+
+    // local plugins are loaded directly, npm packages are loaded
+    // from the plugins directory.
+    const modulePath = local ? path.resolve(process.cwd(), pkg) : this.pluginDir(pkg, version);
+
+    try {
+      return this.require(modulePath, version);
+    } catch (e: any) {
+
+      if (![MODULE_NOT_FOUND_ERROR_CODE, MODULE_VERSION_MISMATCH_ERROR_CODE].includes(e.code)) {
+        // some unexpected error
+        throw e;
+      }
+
+      if (local) {
+        // if a local plugin is missing, nothing we can do about it
+        throw e;
+      }
+
+      // otherwise, we install from npm and re-require.
+      this.installPackage(pkg, version, installEnv);
+      return this.require(modulePath, version);
+    }
+
+  }
+
+  private installPackage(pkg: string, version: string, env: { [key: string]: any }) {
+
+    const pluginDir = this.pluginDir(pkg, version);
+    fs.mkdirpSync(pluginDir);
+
+    const command = [
+      'npm',
+      'install', `${pkg}@${version}`,
+      '--no-save',
+      '--prefix', pluginDir,
+    ].join(' ');
+
+    const finalEnv = { ...process.env };
+    for (const [key, value] of Object.entries(env)) {
+      finalEnv[key] = typeof value === 'string' ? key : JSON.stringify(value);
+    }
+    child.execSync(command, { stdio: ['ignore', 'pipe', 'pipe'], env: finalEnv });
```

Comment:
Might not work properly on Windows if you don't specify `shell: true` (although not sure it applies in this case)

```suggestion
    child.execSync(command, { shell: true, stdio: ['ignore', 'pipe', 'pipe'], env: finalEnv });
```

Reply:
`execSync` doesn't have the `shell` option, it actually always runs it in a shell (unlike `spawnSync`).